apis/telemetry/v1alpha1/shared_types.go

```diff
@@ -29,16 +29,19 @@ type ValueFromSource struct {
 }
 
 func (v *ValueFromSource) IsSecretKeyRef() bool {
-	return v.SecretKeyRef != nil && v.SecretKeyRef.Name != "" && v.SecretKeyRef.Key != ""
+	return v.SecretKeyRef != nil
 }
```

Comment:
The func name was previously already confusing, but this change made it even worse 😄 Maybe let's remove it?